Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -1189,3 +1213,26 @@ extension Sources {
         return swiftSources.count != relativePaths.count
     }
 }
+
+func computeMinimumDeploymentTarget(of binaryPath: AbsolutePath) throws -> PlatformVersion? {
+    let runResult = try Process.popen(arguments: ["xcrun", "vtool", "-show-build", binaryPath.pathString])
+    guard let versionString = try runResult.utf8Output().components(separatedBy: "\n").first(where: { $0.contains("minos") })?.components(separatedBy: " ").last else { return nil }
+    return PlatformVersion(versionString)
+}
+
+private func computeXCTestMinimumDeploymentTarget() -> PlatformVersion {
+    // On macOS, we are determining the deployment target by looking at the XCTest binary.
+    #if os(macOS)
+    do {
+        let runResult = try Process.popen(arguments: ["xcrun", "--sdk", "macosx", "--show-sdk-platform-path"])
+        let sdkPath = AbsolutePath(try runResult.utf8Output().spm_chuzzle() ?? "")
+        let xcTestPath = sdkPath.appending(RelativePath("Developer/Library/Frameworks/XCTest.framework/XCTest"))
+
+        if let version = try computeMinimumDeploymentTarget(of: xcTestPath) {
+            return version
+        }
+    } catch { } // we do not treat this a fatal and instead use the fallback minimum deployment target
```

Comment:
should we emit a warning here?

Reply:
maybe?

Reply:
I think we shouldn't for now. Once we extend this to all platforms as suggested by Jake, we'd always get a warning for watchOS because that code will never succeed there today.

Reply:
This could fail for all kinds of reasons, though, such as if I haven't done `xcode-select` properly, etc.  Could it be confusing to silently get the default fallback deployment target in some cases but not others?